hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/RestRelationships.java

```diff
@@ -164,64 +164,102 @@ public Response getRelationship(@PathParam("relationshipId") String relationship
 
         // delete the relationship
         resolvable.relationships(Relationships.Direction.both).delete(relation.getId());
         if (RestApiLogger.LOGGER.isDebugEnabled()) {
             RestApiLogger.LOGGER.debug("deleting relationship with id: " + relation.getId() + " and name: " +
-                                               relation.getName());
+                    relation.getName());
         }
 
         return Response.noContent().build();
     }
 
     @POST
-    @Path("{path:.*}/relationships")
+    @Path("/relationshipsOf/{entityPath:.*}")
     @ApiOperation("Creates a relationship")
     @ApiResponses({
-            @ApiResponse(code = 201, message = "OK"),
-            @ApiResponse(code = 400, message = "Invalid input data", response = ApiError.class),
-            @ApiResponse(code = 404, message = "Accompanying entity doesn't exist", response =
-                    ApiError.class),
-            @ApiResponse(code = 409, message = "Relationship already exists", response = ApiError.class),
-            @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
-    })
-    public Response create(@PathParam("path") String path,
-                           @ApiParam(required = true) Relationship relation,
+                          @ApiResponse(code = 201, message = "OK"),
+                          @ApiResponse(code = 400, message = "Invalid input data", response = ApiError.class),
+                          @ApiResponse(code = 404, message = "Accompanying entity doesn't exist", response =
+                                  ApiError.class),
+                          @ApiResponse(code = 409, message = "Relationship already exists", response = ApiError.class),
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response create(@ApiParam(required = true) Relationship relation,
                            @Context UriInfo uriInfo) {
 
-        String restPath = fixUpRestPath(path);
         checkForWellKnownLabels(relation.getName(), "create");
-        CanonicalPath cPath = EntityIdUtils.toCanonicalPath(restPath);
+        CanonicalPath cPath = getPath(uriInfo);
         if (!cPath.isDefined()) {
             return Response.status(NOT_FOUND).build();
         }
+
+        @SuppressWarnings("unchecked")
         ResolvableToSingleWithRelationships<Relationship, Relationship.Update> resolvable =
-                getResolvableFromCanonicalPath(cPath);
+                (ResolvableToSingleWithRelationships<Relationship, Relationship.Update>) inventory.inspect(cPath,
+                        ResolvableToSingleWithRelationships.class);
 
         Relationships.Direction directed;
         CanonicalPath theOtherSide;
-        String[] chunks = path.split("/");
-        String currentEntityId = chunks[chunks.length - 1];
+        String currentEntityId = cPath.getSegment().getElementId();
```

Comment:
This is blocker for me, It is not possible to create relationship from tenant to metric/metricType.